### PR TITLE
feat: filter + label notes option

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
- I've just added labeled notes feature + filter option.

### Files to check
- [x] [src/App.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)

#### Component files

- [x] [components/AddEditNote.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-1a3e8710657eca4b2347099df01302643bbea81aab3c53f5b673442a17361000)
- [x] [components/AllNotesFeed.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-5453e30d979e7898e68a41d0c34af9ba200c41d8478c809806158aa1fdda60b2)
- [x] [components/FilterBox.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-287f31da7a0416f853e67ed8b722fd893e1dda92b7ab077716145c5bbfdc294a)
- [x] [components/LabelFeed.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-c48fb2c05e40366c0a2bb67f86846b7075e76e414e916535d1f4cfc5a3e27bdf)
- [x] [components/Note.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-ca4ff88786f04ecde5a7c6f0ad9692a34e8aa9a48912291c9fefaded8d9b082e)

 #### Contexts  + Reducers

- [x] [contexts/FilterContext.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-c7139c611395c3046f6b78b6c63a65e22ea2766a53f2de582c7f4b5afa18dd68)
- [x] [contexts/NoteContext.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-8e8fea03c7d6522afce2ee3fca3792ef83ecd4d5980168f2f0cbfcf9aaf3c43b)
- [x] [reducers/filterReducer.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-9a894a856e4c53d43d41ff13c57f4fa7cf4f4a66cddb1b6572b2b352ba417b3f)

#### utils:
- [x] [/utils/getFilters.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-5377d669886f8d2ad723b207786c9dd56988e9eb65d0488e1a1432237d93a883)
- [x] [/utils/getLabels.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-75acc75bcbd217054e631d851074aef1fca26b37096d28e3ec3b23bf92cc51fc)
- [x] [utils/usePageViewTracker.js](https://github.com/nayyyhaa/NotesFeed/compare/feature-branch?expand=1#diff-22064f7289224f7015c01f0427a1f82942b940f5a5247303fb43784ac0f8b5e7)

#### files to *ignore*
- [ ] files with just few import changed
- [ ] data files
- [ ] CSS files

### Live preview : https://deploy-preview-4--notesfeed.netlify.app/

### CODE REVIEW

**Review by abhisheksahani-nova — APPROVED** [review]
Looks clean , good to merge

File: src/components/FilterBox.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                title="date"
+                checked={sortBy === "OLDTORECENT"}
+                onChange={() => dispatchFilter({ type: "SORT", payload: "OLDTORECENT" })}
+              />
+              Date: Old to recent
+            </label>
+          </li>
+        </ul>
+      </div>
+    </>
```
Comment (ApurvaSawant11) [inline src/components/FilterBox.js]: You already have a `div` enclosing everything. No need for the empty tags

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me. 
Just one suggestion. Align pin and bottom icons in a line. I think it will  look good

image

File: src/components/FilterBox.js (hunk 1 above)
Comment (nayyyhaa) [reply src/components/FilterBox.js]: made the said changes.

**Comment by nayyyhaa** [comment]
> Looks good to me. Just one suggestion. Align pin and bottom icons in a line. I think it will look good
> 
> image

Yes, made the aligned changes.

**Comment by nayyyhaa** [comment]
Thanks @ApurvaSawant11 & @abhisheksahani-nova for reviewing the changes. Merging the PR

**Pull request merged**